File: app/src/main/java/com/pluu/webtoon/item/Episode.java
package com.pluu.webtoon.item;

import android.os.Parcel;
import android.os.Parcelable;

/**
 * 에피소드 Item Class
 * Created by nohhs on 2015-04-06.
 */
public class Episode extends BaseToonInfo implements Parcelable {
	private String episodeId;
	private String episodeTitle;
	private boolean readFlag;

	public Episode(BaseToonInfo info, String episodeId) {
		super(info);
		this.episodeId = episodeId;
	}

	public Episode(Episode item) {
		super(item);
		this.episodeId = item.episodeId;
		this.episodeTitle = item.episodeTitle;
		this.readFlag = item.readFlag;
	}

	public String getEpisodeId() {
		return episodeId;
	}

	public String getEpisodeTitle() {
		return episodeTitle;
	}

	public boolean isReadFlag() {
		return readFlag;
	}

	public void setEpisodeId(String episodeId) {
		this.episodeId = episodeId;
	}

	public void setEpisodeTitle(String episodeTitle) {
		this.episodeTitle = episodeTitle;
	}

	public void setReadFlag() {
		this.readFlag = true;
	}

	@Override
	public int describeContents() { return 0; }

	@Override
	public void writeToParcel(Parcel dest, int flags) {
		super.writeToParcel(dest, flags);
		dest.writeString(this.episodeId);
		dest.writeString(this.episodeTitle);
		dest.writeByte(readFlag ? (byte) 1 : (byte) 0);
	}

	private Episode(Parcel in) {
		super(in);
		this.episodeId = in.readString();
		this.episodeTitle = in.readString();
		this.readFlag = in.readByte() != 0;
	}

	public static final Creator<Episode> CREATOR = new Creator<Episode>() {
		public Episode createFromParcel(Parcel source) {return new Episode(source);}

		public Episode[] newArray(int size) {return new Episode[size];}
	};
}

File: app/src/main/java/com/pluu/webtoon/item/Episode.kt
package com.pluu.webtoon.item

import android.os.Parcel
import android.os.Parcelable

interface IEpisode {
    val webToonId: String
    var episodeId: String
    var episodeTitle: String
}

/**
 * 에피소드 Item Class
 * Created by pluu on 2017-05-04.
 */
class Episode : BaseToonInfo, IEpisode, Parcelable {
    override val webToonId: String = this.toonId
    override var episodeId: String
    override var episodeTitle: String = ""

    var isReadFlag: Boolean = false
        private set

    constructor(info: BaseToonInfo, episodeId: String) : super(info) {
        this.episodeId = episodeId
    }

    constructor(item: Episode) : super(item) {
        episodeId = item.episodeId
        episodeTitle = item.episodeTitle
        isReadFlag = item.isReadFlag
    }

    fun setReadFlag() {
        this.isReadFlag = true
    }

    override fun describeContents() = 0

    override fun writeToParcel(dest: Parcel, flags: Int) {
        super.writeToParcel(dest, flags)
        dest.writeString(this.episodeId)
        dest.writeString(this.episodeTitle)
        dest.writeByte(if (isReadFlag) 1.toByte() else 0.toByte())
    }

    private constructor(source: Parcel) : super(source) {
        episodeId = source.readString()
        episodeTitle = source.readString()
        isReadFlag = source.readByte().toInt() != 0
    }

    companion object {
        @Suppress("unused")
        @JvmField
        val CREATOR: Parcelable.Creator<Episode> = object : Parcelable.Creator<Episode> {
            override fun createFromParcel(source: Parcel) = Episode(source)
            override fun newArray(size: Int): Array<Episode?> = arrayOfNulls(size)
        }
    }
}
